### ISSUE TITLE
Multi-unit synchronous group moving

### ISSUE THREAD

**Issue by SuperiorDonut** [Robosturm/Commander_Wars#1694]
Sometimes when playing I have a large amount of infantry I wish to move forwards in sync, something that becomes _very_ tedious in larger-than-medium maps. For instance, in the attached image I have 19 infantry units that I want to move all 19 of to the right to their furthest potential, and 10 mechs at the top that I wish to move as far to the right as possible. Already a large number, it will continue to grow until I reach near the unit limit cap.

While I can somewhat alleviate this via the multi-turn-plantime function, it has the issue that I can't deliver a command to a location already occupied by an existing unit, thus rendering it non-functional in long "conga line" scenarios; it would be convenient to have a way to move units in synch without needing a unique path for every unit (even if each plan is identical). A way to select a group of units and move them identically would be very convenient for certain scenarios, via some sort of "control select" method followed by issuing the command from a selected unit. 

Some immediately obvious problems come to mind
- what happens when the terrain cost change? Just have them take the path as far as their points allows would probably be best, at least until ->
- what happens when some units would stand on top of others? I imagine stopping the movement one early would be the sensible solution, but then it leads to ->
- how is the order chosen in which units are moved? Does it go from the forward most unit back, highest movement possible first, lowest movement possible first, or some combo? What about age? Oldest first, or newest first? most expensive, or least expensive?
- when some units' paths would go over impassable terrain? And what about traps? Probably just stop the unit immediately and let the rest move after
- would this become computationally expensive? I assume if every unit takes the identical pathing, follows a standard order (expensive->cheap, oldest->newest, then highest move speed or "worst travel type" (tire A->tire B->hovercraft->treads->feet/crawler->mech->air/boat), with triple-ties broken by dice rolls), and goes the furthest possible distance in order, and doesn't take any rerouting, that it wouldn't be too complex computation wise (the many drawn arrows aside, perhaps)
- what about different movement distances? I would imagine you would be able to move the group as far as the highest-travel-potential-unit can go, but stop each units pathing along their typical distance movements.
- how would the animations for this work? I would imagine drawing 10s of arrows and overlapping movement diamonds simultaneously could be strenuous; perhaps a simplified "only shows the arrow of the chosen unit, given the movement options of the furthest-possible-path of a unit in the selection, and leave the rest to be executed without a "resulting example" (which could be done via the sprites disappearing and blinking at their resulting position (to avoid "same-resulting-spot confusion"), if one were to be added)
- what if I want to issue a capture command to a certain unit in the group? The best way around imo (and a QoL feature that should probably be added outside of this anyways) would be simply to have it so that any infantry that waits on a neutral city automatically issues a capture command. I was playing multiplayer with a first-time-advance-wars-playing friend today, and he made such an error multiple times; you already automatically capture cities when ending a turn with a ready-to-move unit that has already begun a capture sequence, and I can't imagine scenarios where you *wouldn't* want to begin capturing a city you wait a foot-unit on, except to save it for an ally in online 2v2 (and similar) multiplayer games (which is too niche imo, and could be a QoL feature turned off for ally-having multiplayer games)
- likely other issues I cannot think about remain, too

Allowing the selection of same-type units that have universal-1-movement-costs (aka, mechs and air-units, except when snowing) seems like something that would be very feasible to add though, and to a lesser extent infantry/ships (as those units don't have any impassable terrain aside from sea/land respectively).

This seems like a lot of work with many potential edge cases though, so I would understand if this is "definite no," let alone a "low priority/post-beta" addition. It still would make my gameplay experience far less tedious when in large maps though, so I feel like a feature request post makes sense regardless.

Thanks for reading :)

![Annoying infantry](https://github.com/Robosturm/Commander_Wars/assets/114554928/fa6ae4db-5e55-479d-8b34-fba40a719d36)

**Comment by Robosturm** [comment]
I see a lot more problems with that while you can get a solution. I would say that the complexity overhead realizing this featue out class it's benefit. Since it's only a useful feature on large maps which are a niche part of AW/CoW. 
TBS are very different in that aspect to RTS games where a multi move feature is a must and comes with nearly no drawbacks

**Comment by TheMasterCreed** [comment]
The only solution I can think of is batch queueing which is a much more practical fix and accomplishes what you are asking for
@SuperiorDonut

**Comment by Robosturm** [comment]
Should be improved with the quick movement feature via a second button
